src/test/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImplTest.scala

```diff
@@ -7,94 +7,205 @@ import mesosphere.marathon.core.matcher.base.OfferMatcher
+      dummySource, task, MarathonTestHelper.makeTaskFromTaskInfo(task)))
     val offerProcessor = createProcessor()
 
     val deadline: Timestamp = clock.now() + 1.second
-    when(offerMatcher.matchOffer(deadline, offer)).thenReturn(
-      Future.successful(MatchedTasks(offerId, tasksWithSource))
-    )
-    when(taskLauncher.launchTasks(offerId, tasks)).thenReturn(true)
 
+    And("an cooperative offerMatcher and taskTracker")
+    offerMatcher.matchOffer(deadline, offer) returns Future.successful(MatchedTasks(offerId, tasksWithSource))
+    for (task <- tasksWithSource) {
+      taskTracker.store(appId, task.marathonTask) returns Future.successful(mock[PersistentEntity])
+    }
+
+    And("a working taskLauncher")
+    taskLauncher.launchTasks(offerId, tasks) returns true
+
+    When("processing the offer")
     Await.result(offerProcessor.processOffer(offer), 1.second)
 
-    verify(offerMatcher).matchOffer(deadline, offer)
-    verify(taskLauncher).launchTasks(offerId, tasks)
+    Then("we saw the offerMatch request and the task launches")
+    verify(offerMatcher, times(1)).matchOffer(deadline, offer)
+    verify(taskLauncher, times(1)).launchTasks(offerId, tasks)
 
+    And("all task launches have been accepted")
     assert(dummySource.rejected.isEmpty)
     assert(dummySource.accepted.toSeq == tasks)
+
+    And("the tasks have been stored")
+    for (task <- tasksWithSource) {
+      verify(taskTracker, times(1)).created(appId, task.marathonTask)
+      verify(taskTracker, times(1)).store(appId, task.marathonTask)
+    }
   }
 
   test("match successful, launch tasks unsuccessful") {
+    Given("an offer")
+    val dummySource = new DummySource
+    val tasksWithSource = tasks.map(task => TaskWithSource(
+      dummySource, task, MarathonTestHelper.makeTaskFromTaskInfo(task)))
+
+    val offerProcessor = createProcessor()
+
+    val deadline: Timestamp = clock.now() + 1.second
+    And("an cooperative offerMatcher and taskTracker")
+    offerMatcher.matchOffer(deadline, offer) returns Future.successful(MatchedTasks(offerId, tasksWithSource))
+    for (task <- tasksWithSource) {
+      taskTracker.store(appId, task.marathonTask) returns Future.successful(mock[PersistentEntity])
+      taskTracker.terminated(appId, task.marathonTask.getId) returns Future.successful(Some(task.marathonTask))
+    }
+
+    And("a dysfunctional taskLauncher")
+    taskLauncher.launchTasks(offerId, tasks) returns false
+
+    When("processing the offer")
+    Await.result(offerProcessor.processOffer(offer), 1.second)
+
+    Then("we saw the matchOffer request and the task launch attempt")
+    verify(offerMatcher, times(1)).matchOffer(deadline, offer)
+    verify(taskLauncher, times(1)).launchTasks(offerId, tasks)
+
+    And("all task launches were rejected")
+    assert(dummySource.accepted.isEmpty)
+    assert(dummySource.rejected.toSeq.map(_._1) == tasks)
+
+    And("the tasks where first stored and then expunged again")
```

Comment:
It sounds like you care about the order of the following calls.

Mockito doesn't care about the order of the calls unless you use an inOrder object.